sqlglot/parser.py

```diff
@@ -1109,6 +1109,9 @@ class Parser(metaclass=_Parser):
     # Whether or not interval spans are supported, INTERVAL 1 YEAR TO MONTHS
     INTERVAL_SPANS = True
 
+    # Whether the dialect allows syntax such as: select * from t1 partition (p0)
```

Comment:
Sounds good. 

Testing the MySQL codepath for RTT should yield the same query back in both main and in this PR, but the AST is different. Do you mean to test it's partition selection by somehow checking the `partition` arg in `exp.Table`?

Reply:
How is it different? You mean now we parse an actual partition clause instead of the alias, right? If so, then for

> Do you mean to test it's partition selection by somehow checking the partition arg in exp.Table?

A simple validate identity will suffice (since we won't have the `AS ...` now)